### PR TITLE
Xcm docs

### PR DESCRIPTION
**Pull request by albertov19**
This PR adds the first draft of the XCM-overview page. This section should be expanded to include all the documentation about XCM (not tutorials)

### CODE REVIEW

File: learn/xcm/overview.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+Furthermore, the two most common use-cases for XCM messages, at least in the early stages of its implementations, are:
+
+- **Asset Teleporting** — consists of moving an asset from one blockchain to another by destroying the amount being transferred in the origin chain and creating a clone (same amount as destroyed) on the target chain. In such cases, each chain holds the native asset as reserve, similar to a burn-mint bridging mechanism. The model requires a certain degree of trust, as any of the two chains could maliciously mint more assets
+- **Remote Transfers** — consists of moving an asset from one blockchain to another via an intermediate account in the origin chain that is trustlessly owned by the target chain. This intermediate account is known as the "Sovereign" account. In such cases, the origin chain asset is not destroyed but held by the Sovereign account. The XCM execution in the target chain mints a wrapped (referred to virtual) representation to a target address. The wrapped (virtual) representation is always interchangeable on a 1:1 basis with the native asset. This is similar to a lock-mint / burn-unlock bridging mechanism
```
Comment (themacexpert) [inline learn/xcm/overview.md:30]: (referred to _as_ virtual)

File: learn/xcm/overview.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    * Maximum size of the messages to be sent
+
+The transaction fees are paid in the cross-chain (xc) representation of the relay chain asset (_xcRelayChainAsset_). For example, for Kusama/Moonriver, the transaction fees would be paid in _xcKSM_. Therefore, the account paying for the fees must have enough _xcRelayChainAsset_. This might be tackled in Moonbeam/Moonriver by having fees from incoming XCM messages, which are paid in the origin chain asset, sent to the treasury, and using the treasury account to pay for the channel registration extrinsic.
+
+Even though parachain A has expressed its intentions of opening and XCM channel with parachain B, the latter has not signaled to the relay chain its intentions to receive messages from parachain A. Therefore, to have an established channel, parachain B must also send an extrinsic (which is also an XCM) to the relay chain. The accepting channel extrinsic is similar to the previous one. However, the encoded call data only includes the new method (accept channel) and the parachain ID of the sender (parachain A in this example). Once both parachains have agreed, the channel is opened within the next epoch change.
```
Comment (themacexpert) [inline learn/xcm/overview.md:57]: opening _an_ XCM channel

File: learn/xcm/overview.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+Expanding on Moonbeam's unique Ethereum compatibility features, foreign assets will be represented via a standard [ERC-20 interface](https://github.com/PureStake/moonbeam/blob/master/precompiles/assets-erc20/ERC20.sol){target=_blank} through a precompiled contract. XCM assets on Moonbeam are called XC-20s, to differentiate native XCM assets from ERC-20 generated via the EVM. The precompile contract will access the necessary Substrate functions to perform the required actions. Nevertheless, from a developer's perspective, XC-20s are ERC-20 tokens with the added benefit of being a XCM cross-chain asset, and dApps can easily support them through a familiar ERC-20 interface. 
+
+![Moonbeam XC-20 XCM Integration With Polkadot](/images/learn/xcm/overview-4.png)
+
+The precompile itself does not support cross-chain transfers to stay as close as possible to the original ERC-20 interface. Consequently, developers will have to rely on the Substrate API and XCMs to move the assets back to their original chain, or on a different [precompile contract](TODO) to access such features from the Ethereum API.
```
Comment (themacexpert) [inline learn/xcm/overview.md:92]: Friendly reminder to update the precompile contract link.

File: learn/xcm/overview.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+Even though parachain A has expressed its intentions of opening and XCM channel with parachain B, the latter has not signaled to the relay chain its intentions to receive messages from parachain A. Therefore, to have an established channel, parachain B must also send an extrinsic (which is also an XCM) to the relay chain. The accepting channel extrinsic is similar to the previous one. However, the encoded call data only includes the new method (accept channel) and the parachain ID of the sender (parachain A in this example). Once both parachains have agreed, the channel is opened within the next epoch change.
+
+![XCM Channel Registration Overview](/images/learn/xcm/overview-3.png)
```
Comment (themacexpert) [inline learn/xcm/overview.md:59]: I'm not really a fan of the Moonbeam logo on top of the person to represent the Sovereign account. To me it just looks like a regular user in the logo. Can we change it to something that more represents a group / sovereign entity such as a democracy / council / state related logo?

**Review by themacexpert — APPROVED** [review]

File: learn/xcm/overview.md (hunk 4 above)
Comment (albertov19) [reply learn/xcm/overview.md:59]: The thing is that it is an account owned by the Moonbeam parachain

**Pull request merged**